addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1361,3 +1412,108 @@ func (r *ClusterBootstrapReconciler) reconcileClusterProxyAndNetworkSettings(clu
 
 	return nil
 }
+
+func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, log logr.Logger) (ctrl.Result, error) {
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	err = remoteClient.Get(r.context, client.ObjectKeyFromObject(cluster), clusterBootstrap)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if clusterBootstrap == nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.removeAdditionalPackageInstalls(cluster, log)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	packageInstallsArePresent := hasPackageInstalls(r.context, remoteClient, cluster, r.Config.SystemNamespace, clusterBootstrap.Spec.AdditionalPackages, log)
```

Comment:
Is this check meant to be wrapped in a retry loop?

Reply:
timeOutReached is the max time we want to spend pausing the deletion of the cluster. 
Once that time is reached we go ahead and continue deleting regardless of weather or not packageInstall crs are present. 
Deleting the packageInstall CRs could take quite a bit of time. We could sit here and continuously retry, but perhaps is best to just reque with a requeAfter,  let the system do something els, and come back again and check the timeout before we make any more calls to the api, since regardless of what the api says, we will delete cluster after timeOutReached = true.